specs/agents/metadata.md

```diff
@@ -24,42 +24,38 @@ System metadata relates to the host/container in which the service being monitor
 
 #### Hostname
 
-This hostname reported by the agent is mapped by the APM Server to the 
-[`host.hostname` ECS field](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-hostname), which should 
-typically contain what the `hostname` command returns on the host machine. However, since we rely on this field for 
-our integration with beats data, we should attempt to follow a similar logic to the `os.Hostname()` Go API, which beats 
-relies one. While `os.Hostname()` contains some complex OS-specific logic to cover all sorts of edge cases, our 
-algorithm should be simpler. It relies on the execution of external commands with a fallback to standard environment 
-variables. Agents SHOULD implement this hostname discovery algorithm wherever possible:
+The hostname value(s) reported by the agent are mapped by APM Server to the ECS
+[`host.hostname`](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-hostname) and
+[`host.name`](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-name) fields.
+
+Agents SHOULD return the lower-cased FQDN whenever possible, which might require a DNS query.
+
+Agents SHOULD implement this hostname discovery algorithm wherever possible:
 ```
 var hostname;
 if os == windows
-  hostname = exec "cmd /c hostname"                   // or any equivalent *
+  // https://stackoverflow.com/questions/12268885/powershell-get-fqdn-hostname
+  // https://learn.microsoft.com/en-us/dotnet/api/system.net.dns.gethostentry
+  hostname = exec "powershell.exe [System.Net.Dns]::GetHostEntry($env:computerName).HostName" // or any equivalent *
+  if (hostname == null || hostname.length == 0)
+    hostname = exec "cmd.exe /c hostname"               // or any equivalent *
   if (hostname == null || hostname.length == 0)
     hostname = env.get("COMPUTERNAME")
 else 
-  hostname = exec "uname -n"                          // or any equivalent *
-  if (hostname == null || hostname.length == 0)
-    hostname = exec "hostname"                        // or any equivalent *
+  hostname = exec "hostname -f"                         // or any equivalent *
```

Comment:
@SylvainJuge In my testing while implementing for .NET, it seems that this command may return with a period suffix (e.g. `DESKTOP-NCGCCT0.` or even the suffix `.localdomain`. Do we send as-is, or should we consider trimming in either case?

Reply:
@stevejgordon Where did you see an example like `DESKTOP-NCGCCT0.`? Is that an OS other than Windows?

> or even the suffix .localdomain

I'm not sure if it is equivalent, but on macOS the domain is commonly ".local", e.g. on my laptop:

```
% hostname -f
pink.local
```

Reply:
So I'm testing using WSL2 on Windows with Ubuntu. On my Windows laptop, where no domain is set, it reports the period suffix when I run `hostname -f` from Ubuntu (using WSL2). I also set up a Windows 11 VM to test adding the DNS suffix and again added WSL Ubuntu to test the code and picked up the FQDN, which it did. Once I removed the DNS suffix, `.localdomain` started appearing.

Reply:
For .NET, as the primary mechanism, I'm planning to use the `Dns.GetHostEntry(string.Empty).HostName` API, which is cross-platform aware and seems to potentially include the suffix, at least when testing on Ubuntu via WSL2.

Reply:
I think here that's clearly a case where using the API seems to be more consistent than the equivalent with `hostname`.
Does it also returns a consistent value when invoked outside of WSL ?

Reply:
The Windows API here is what I should use for consistency and to avoid any external process calls. Here are what it produces in my testing.

```
WITHOUT EVER SETTING DOMAIN SUFFIX

WINDOWS: WinDev2306Eval
LINUX (WSL): WinDev2306Eval.

WITH DOMAIN SUFFIX

WINDOWS: WinDev2306Eval.test.domain
LINUX (WSL): WinDev2306Eval.test.domain

AFTER REMOVING DOMAIN SUFFIX SET

WINDOWS: WinDev2306Eval
LINUX (WSL): WinDev2306Eval.localdomain
```

Reply:
On reflection, it probably makes sense not to try and trim this and accept the OS-specific result as it's consistent and easy to document that our detected utilizes a Microsoft API. Consumers can always provide their preferred name if they don't like the detected one.

Reply:
@stevejgordon Assuming I followed the code correctly, Beats are currently using https://github.com/elastic/go-sysinfo/blob/9ea2eba5301c77b511c0475d7137ea3437a55a9a/providers/windows/host_windows.go#L87-L94
(https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getcomputernameexa)

Reply:
@trentm The .NET API eventually seems to calling into the native Windows Sockets 2, Winsock API `gethostname` function (https://learn.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-gethostname) and on Linux the `gethostname` system call.

Reply:
So, if I understand this correctly, you are suggesting to just remove the extra `.` that is appended when calling this API through WSL ?

Reply:
@SylvainJuge Originally, yes, I thought perhaps we should. However, as it's consistent with `hostname -f` and therefore, presumably what other agents on the same host will send, perhaps best to leave it unchanged.

Reply:
@stevejgordon do you think we need to add a clarification about this in the spec or could we leave it as-is ?